ddtrace/tracer/span_test.go

```diff
@@ -204,6 +204,27 @@ func TestSpanFinishWithErrorStackFrames(t *testing.T) {
 	assert.Equal(strings.Count(span.Meta[ext.ErrorStack], "\n\t"), 2)
 }
 
+// nilStringer is used to test nil detection when setting tags.
+type nilStringer struct {
+	s string
+}
+
+// String incorrectly assumes than n will not be nil in order
+// to ensure SetTag identifies nils.
+func (n *nilStringer) String() string {
+	return n.s
+}
+
+type panicStringer struct {
```

Comment:
Perhaps you can merge the two into a `panickyStringer` and just add a bool inside the struct, something like `willPanic` and then have that define whether the `String` call panics.

Reply:
Since the `panickyStringer` would be `nil`, we can't reference the boolean inside it to check which way we need to panic.